File: frontend/app/src/main/java/com/example/nomo/utils/Validators.java
package com.example.nomo.utils;

public class Validators {

    public static boolean isValidNickname(String nickname) {
        return nickname != null && nickname.length() >= 3 && nickname.length() <= 20;
    }

    public static boolean isValidEmail(String email) {
        return android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
    }

    public static boolean isValidPassword(String password) {
        return password != null && password.length() >= 6;
    }
}

File: frontend/app/src/main/java/com/example/nomo/utils/Validators.kt
package com.example.nomo.utils

import android.util.Patterns

object Validators {
    fun isValidEmail(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }

    fun isValidPassword(password: String): Boolean {
        return password.length >= 6
    }

    fun isValidNickname(nickname: String): Boolean {
        return nickname.length in 3..20
    }
}
